File: JSPM.Web/Modules/ServerTypes/Students/TeacherEndorsementForm.ts
import { StringEditor, IntegerEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface TeacherEndorsementForm {
    TeacherId: StringEditor;
    StudentId: IntegerEditor;
    Title: StringEditor;
    Details: StringEditor;
}

export class TeacherEndorsementForm extends PrefixedContext {
    static formKey = 'Students.TeacherEndorsement';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!TeacherEndorsementForm.init)  {
            TeacherEndorsementForm.init = true;

            var w0 = StringEditor;
            var w1 = IntegerEditor;

            initFormType(TeacherEndorsementForm, [
                'TeacherId', w0,
                'StudentId', w1,
                'Title', w0,
                'Details', w0
            ]);
        }
    }
}

File: JSPM.Web/Modules/Students/TeacherEndorsement/TeacherEndorsementForm.cs
using Serenity.ComponentModel;

namespace JSPM.Students.Forms;

[FormScript("Students.TeacherEndorsement")]
[BasedOnRow(typeof(TeacherEndorsementRow), CheckNames = true)]
public class TeacherEndorsementForm
{
    public int TeacherId { get; set; }
    public int StudentId { get; set; }
    public string Title { get; set; }
    public string Details { get; set; }
}